Use rancher catalog

adamgraves-choices opened the pull request.

adamgraves-choices committed:
Update to rancher-cli 0.5.0

adamgraves-choices committed:
Use rancher catalog install

neilbmclaughlin commented:
@adamgraves-choices - I realise it's still WIP but was interested in it and it looks good.

I had a play wth the rancher API last week and came up with the following snippets which might simplify the script:

_Get the latest version version id:_

`RANCHER_TEMPLATE_VERSION=$(curl -su "${RANCHER_ACCESS_KEY}:${RANCHER_SECRET_KEY}" "${RANCHER_URL}/v1-catalog/templates/${CATALOG_ID}:${RANCHER_TEMPLATE_NAME}" | jq --raw-output '.defaultTemplateVersionId')`

**Note**: I've assumed that default and latest are the same thing. Not sure if there is a scenario where default might not be the latest. To get a feel for the data being returned just do the above but change the jq filter to `jq '.'`

_Get the rancher compose file:_

`curl  -su "${RANCHER_ACCESS_KEY}:${RANCHER_SECRET_KEY}" "${RANCHER_URL}/v1-catalog/templates/${RANCHER_TEMPLATE_VERSION} | jq --raw-output '.files["rancher-compose.yml"]' > rancher-compose.yml`

adamgraves-choices committed:
Use Catalog for version and compose file

adamgraves-choices committed:
Use docker to run rancher

adamgraves-choices committed:
Use docker for yml2json function

neilbmclaughlin commented on rancher_pr_deploy.sh:
```diff
@@ -55,14 +55,18 @@ rancher() {
   curl -su "${RANCHER_ACCESS_KEY}:${RANCHER_SECRET_KEY}" "${RANCHER_SERVER}/v1-catalog/templates/${RANCHER_CATALOG_ID}" | \
   jq --raw-output '.files["rancher-compose.yml"]' | \
-    python -c 'import sys, yaml, json; json.dump(yaml.load(sys.stdin), sys.stdout)' | \
+    yml2json | \
```
This means we no longer need the `pip install` line in the .travis.yml file

neilbmclaughlin replied:
Also, not sure how you weigh up the benefits of running a single command (python/rancher/etc) in a docker container vs installing it in Travis. I guess (as you mentioned in the catchup) it makes the script more self contained and able to be run in environments other than Travis?

neilbmclaughlin replied:
I which case should we be doing the same for docker-compose?

neilbmclaughlin replied:
Finally, yml2json is broken. The -a flag needs STDIN/STDOUT and is redundent I think. Played around trying to get it to work for a bit but with no luck (the python yaml package needs installing in the container).

adamgraves-choices replied:
We don't need to convert the YAML to json anymore anyway, so it's a moot point.

neilbmclaughlin committed:
:bug: Faiul build when rancher catalog install fails

neilbmclaughlin commented:
Running the command through shellcheck gave several warnings. Also, been bugging me the deployment to rancher didn't previously break the build. I've created a PR which resolves these issues here #26.

adamgraves-choices committed:
Remove need for yaml conversion

adamgraves-choices committed:
Merge branch 'feature/use_rancher_catalog' into fix/error-when-rancher-catalog-install-fails

adamgraves-choices committed:
Merge pull request #26 from nhsuk/fix/error-when-rancher-catalog-install-fails

Fail build when rancher catalog install fails

adamgraves-choices commented:
Cheers @neilbmclaughlin, merged. Removed the `yaml2json` function as we can get the questions directly from the catalog API.

I've removed the WIP label, so when you're happy it's good to go.
